system/jlib/jutil.cpp

```diff
@@ -812,27 +814,157 @@ class CRandom: public CInterface, public IRandomNumberGenerator
         return ret;
     }
 
-} RandomMain; 
+};
 
-static CriticalSection gobalRandomSect;
+thread_local Owned<IRandomNumberGenerator> RandomMain = createRandomNumberGenerator();
 
 unsigned getRandom()
 {
-    CriticalBlock block(gobalRandomSect); // this is a shame but it is not thread-safe without
-    return RandomMain.next();
+    return RandomMain->next();
```

Comment:
This is changing the implementation of semantics of an existing function.  It would need careful checking - e.g., how are they seeded?